stdlib/InteractiveUtils/src/InteractiveUtils.jl

```diff
@@ -49,17 +52,20 @@ function varinfo(m::Module=Main, pattern::Regex=r""; all::Bool = false, imported
             if recursive && !isbuiltin && isa(value, Module) && value !== m2 && nameof(value) === v && parentmodule(value) === m2
                 push!(workqueue, (value, "$prep$v."))
             end
-            ssize_str, ssize = if isbuiltin
-                    ("", typemax(Int))
-                else
-                    ss = summarysize(value)
-                    (format_bytes(ss), ss)
+            Threads.@spawn begin
+                ssize_str, ssize = if isbuiltin
+                        ("", typemax(Int))
+                    else
+                        ss = summarysize(value)
+                        (format_bytes(ss), ss)
+                    end
+                if ssize >= minsize
+                    push!(rows_per_thread[Threads.threadid()], Any[string(prep, v), ssize_str, summary(value), ssize])
```

Comment:
I strongly recommend avoiding `buffers[Threads.threadid()]` approach. In particular, this code is not data-race-free anymore since it is perfectly fine to do I/O in `summary` (it's a bit strange, but it's possible). That is to say, we can't assume

```julia
tid = threadid()
summary(x)
@assert tid == threadid()
```

anymore due to task migration.

If you *really* want to do it, you can use

```julia
r = Any[string(prep, v), ssize_str, summary(value), ssize]
push!(rows_per_thread[Threads.threadid()], r)
```

so that no I/O is executed after `rows_per_thread[Threads.threadid()]` is retrieved and before its mutated. But as you can see, it's very tricky to reason about what is OK and what is not. (Also, in a way it is "accidentally OK" since we don't say that `push!(::Vector, _)` doesn't do I/O. It's almost surely won't, but somebody may write a compiler plugin to insert an I/O at every call for logging or something.)

Reply:
d'oh, I keep forgetting about task migration.. I'll drop the last commit